src/network/serverpackethandler.cpp

```diff
@@ -1359,22 +1359,24 @@ void Server::handleCommand_Interact(NetworkPacket* pkt)
 		(only when digging or placing things)
 	*/
 	if (action == 0 || action == 2 || action == 3) {
-		float d = player_pos.getDistanceFrom(pointed_pos_under);
-		float max_d = BS * 14; // Just some large enough value
-		if (d > max_d) {
-			actionstream << "Player " << player->getName()
-					<< " tried to access " << pointed.dump()
-					<< " from too far: "
-					<< "d=" << d <<", max_d=" << max_d
-					<< ". ignoring." << std::endl;
-			// Re-send block to revert change on client-side
-			RemoteClient *client = getClient(pkt->getPeerId());
-			v3s16 blockpos = getNodeBlockPos(floatToInt(pointed_pos_under, BS));
-			client->SetBlockNotSent(blockpos);
-			// Call callbacks
-			m_script->on_cheat(playersao, "interacted_too_far");
-			// Do nothing else
-			return;
+		if (!isSingleplayer() && !g_settings->getBool("disable_anticheat")) {
```

Comment:
Additionally, this conditional could be combined with the action checking conditional.  It would reduce indentation and make the diff much smaller, making it easier to understand.

Also, note that this code is over-simplified.  If you have an item with a reach distance of 14 or more it'll be considered cheaty.